sky/provision/azure/instance.py

```diff
@@ -311,29 +311,8 @@ def _create_vm(
         vm_name=vm_name,
         parameters=vm_instance,
     )
-    # poller.result() will block on async operation until it's done.
+    vm_poller.wait()
     logger.info(f'Created VM {vm_poller.result().name}.')
```

Comment:
Just curious, why don't set the result to a variable and reuse them later?
```suggestion
    # Wait for the creation of VM and get the result.
    vm_poller_result = vm_poller.result()
    logger.info(f'Created VM {vm_poller_result.name}.')
```

Reply:
No strong reasons, just having less code